src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+`Employee`.
+
+==== Fields
+
+The engine reads the first local entry of the `IMap` to get the list of fields
+accessible from the SQL. The engine extracts field metadata (name, type)
+from the key and the value, and combines them into a final list of fields.
+
+Field metadata extraction logic depends on the serialization format, as described
+below.
+
+-  For Java objects (`Serializable`, `DataSerializable`, `IdentifiedDataSerializable`),
+the object is deserialized if needed and then analyzed using reflection. Only public
+fields and getters are taken in count. A public field is exposed using its exact name.
+A getter is exposed using the JavaBean naming conventions. E.g. `getName` is exposed as
+`name` and `isActive` is exposed as `active`
+- For `Portable` objects, only the fields that are written in the `Portable.writePortable`
+method are exposed using their exact names
+- `HazelcastJsonValue` objects cannot be queried at the moment. This limitation will be
+removed in future releases
+
+Field names are case-sensitive.
+
+If the `IMap` does not have local entries on the member where the query is started,
+then the list of fields cannot be resolved, and an exception is thrown.
+
+We refer to the key and the value as *top-level* objects. The key is exposed as
+`__key` field, and the value is exposed as `this` field.
+
+If a top-level object does not have nested fields, then it is exposed as a normal
+field. Otherwise, it is exposed a *hidden field*. Hidden fields behave similarly
+to normal fields, but are not returned by `SELECT * ...` queries.
+
+If both the key and the value have fields with the same name, then the field of the
+value will be used.
+
+==== Indexes
+
+The engine could use `IMap` indexes to speed up the execution of certain queries.
+
+`SORTED` and `HASH` indexes are supported.
+
+==== HD
+
+`IMap` objects with the `InMemoryFormat.NATIVE` could only be queried if they have
+at least one index, and the property `hazelcast.hd.global.index.enabled` is set to
+`true`.
+
+=== Supported Features
+
+==== Data Types
+
+The following SQL types are currently supported.
+
+General:
+
+- `BOOLEAN` (`java.lang.Boolean`)
+- `VARCHAR` (`java.lang.String`, `java.lang.Character`)
+- `OBJECT`
+
+Numeric:
+
+- `TINYINT` (`java.lang.Byte`)
+- `SMALLINT` (`java.lang.Short`)
+- `INTEGER` (`java.lang.Integer`)
+- `BIGINT` (`java.lang.Long`)
+- `DECIMAL` (`java.math.BigDecimal`, `java.math.BigInteger`)
+- `REAL` (`java.lang.Float`)
+- `DOUBLE` (`java.lang.Double`)
+
+Temporal:
+
+- `DATE` (`java.time.LocalDate`)
+- `TIME` (`java.time.LocalTime`)
+- `TIMESTAMP` (`java.time.LocalDateTime`)
+- `TIMESTAMP_WITH_TIME_ZONE` (`java.time.OffsetDateTime`, `java.time.ZonedDateTime`,
+`java.time.Instant`, `java.util.Date`, `java.util.Calendar`)
+
+==== Operators
+
+It is possible to execute project-filter queries in the form
+`SELECT ... FROM <map_name> WHERE ...` on a single map.
+
+The support for the following operators are planned for future releases:
```

Comment:
Let's be more explicit. With bold font, say something like "following operators ARE NOT supported in this release. The support is planned for future releases".

Reply:
Added an explicit note that features are not supported.